spec/classes/mod/authnz_ldap_spec.rb

```diff
@@ -41,50 +41,53 @@
   end # Debian
 
   context 'default configuration with parameters on a RedHat OS' do
-    let :facts do
-      {
-        osfamily: 'RedHat',
-        operatingsystemrelease: '6',
-        id: 'root',
-        kernel: 'Linux',
-        operatingsystem: 'RedHat',
-        path: '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
-        is_pe: false,
-      }
-    end
-
-    it { is_expected.to contain_class('apache::params') }
-    it { is_expected.to contain_class('apache::mod::ldap') }
-    it { is_expected.to contain_apache__mod('authnz_ldap') }
-
-    context 'default verify_server_cert' do
-      it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert On$}) }
-    end
-
-    context 'verify_server_cert = false' do
-      let(:params) { { verify_server_cert: false } }
-
-      it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert Off$}) }
-    end
-
-    context 'verify_server_cert = wrong' do
-      let(:params) { { verify_server_cert: 'wrong' } }
-
-      it 'raises an error' do
-        expect { is_expected.to raise_error Puppet::Error }
+    on_supported_os.each do |os, facts|
+      next unless os.start_with? 'redhat'
+      context "On #{os}" do
+        let :facts do
+          facts
+        end
+
+        it { is_expected.to contain_class('apache::params') }
+        it { is_expected.to contain_class('apache::mod::ldap') }
+        it { is_expected.to contain_apache__mod('authnz_ldap') }
+
+        if facts[:os]['version'].to_i >= 7
+          it { is_expected.to contain_package('mod_authz_ldap') }
+        else
+          it { is_expected.to contain_package('mod_ldap') }
+        end
+
+        context 'default verify_server_cert' do
+          it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert On$}) }
+        end
+
+        context 'verify_server_cert = false' do
+          let(:params) { { verify_server_cert: false } }
+
+          it { is_expected.to contain_file('authnz_ldap.conf').with_content(%r{^LDAPVerifyServerCert Off$}) }
+        end
+
+        context 'verify_server_cert = wrong' do
+          let(:params) { { verify_server_cert: 'wrong' } }
+
+          it 'raises an error' do
+            expect { is_expected.to raise_error Puppet::Error }
```

Comment:
I know it was there before, but you may want to use a more descriptive syntax:
```suggestion
            is_expected.to compile.and_raise_error
```